lib/scala/distribution-manager/src/main/scala/org/enso/distribution/config/GlobalConfig.scala

```diff
@@ -66,6 +71,106 @@ object GlobalConfig {
+          }
+          val (nameOpt, emailOpt) = bindings.get("author") match {
+            case Some(authorNode: MappingNode) =>
+              val authorBindings = mappingKV(authorNode)
+              (
+                authorBindings
+                  .get("name")
+                  .map(stringDecoder.decode)
+                  .getOrElse(Right(None))
+                  .asInstanceOf[Either[Throwable, Option[String]]],
+                authorBindings
+                  .get("email")
+                  .map(stringDecoder.decode)
+                  .getOrElse(Right(None))
+                  .asInstanceOf[Either[Throwable, Option[String]]]
+              )
+            case _ =>
+              // Fallback
+              (
+                bindings
+                  .get(Fields.AuthorName)
+                  .map(stringDecoder.decode(_).map(Some(_)))
+                  .getOrElse(Right(None)),
+                bindings
+                  .get(Fields.AuthorEmail)
+                  .map(stringDecoder.decode(_).map(Some(_)))
+                  .getOrElse(Right(None))
+              )
+          }
+          val editionProviderOpt = bindings
+            .get(Fields.EditionProviders)
+            .map(seqStringDecoder.decode)
+            .getOrElse(Right(Seq.empty))
+          for {
+            defaultVersion  <- defaultVersionOpt
+            name            <- nameOpt
+            email           <- emailOpt
+            editionProvider <- editionProviderOpt
+          } yield GlobalConfig(defaultVersion, name, email, editionProvider)
+      }
+    }
+
+  implicit val encoderSnake: SnakeYamlEncoder[GlobalConfig] =
+    new SnakeYamlEncoder[GlobalConfig] {
+      override def encode(value: GlobalConfig): AnyRef = {
+        val defaultVersionEncoder = implicitly[SnakeYamlEncoder[DefaultVersion]]
+        val editionProviders      = implicitly[SnakeYamlEncoder[Seq[String]]]
+        val elements              = new util.ArrayList[(String, AnyRef)]()
+        elements.add(
+          (
+            "default",
+            toMap(
+              "enso-version",
+              defaultVersionEncoder.encode(value.defaultVersion)
+            )
+          )
+        )
+        if (value.authorName.nonEmpty || value.authorEmail.nonEmpty) {
+          val authorElements = new util.ArrayList[(String, AnyRef)]()
+          value.authorName.foreach(v => authorElements.add(("name", v)))
+          value.authorEmail.foreach(v => authorElements.add(("email", v)))
+          elements.add(("author", toMap(authorElements)))
+        }
+        if (value.editionProviders.nonEmpty) {
+          elements.add(
+            (
+              Fields.EditionProviders,
+              editionProviders.encode(value.editionProviders)
+            )
+          )
+        }
+        toMap(elements)
+      }
+    }
+
   /** [[Decoder]] instance for [[GlobalConfig]].
     */
   implicit val decoder: Decoder[GlobalConfig] = { json =>
```

Comment:
Why do we still keep `io.circe` encoder here? So this `GlobalConfig` class is encoded both in yaml and in json? Where is the JSON encoding still needed from?

Reply:
It's possible this particular one can be removed, let me check. There is currently some duplication, e.g. in `ComponentGroups` because we also send those over the wire and need them for JSON responses.

Reply:
This one could be removed. Others not so much.